src/Gaufrette/Adapter/AzureBlobStorage.php

```diff
@@ -47,22 +49,52 @@ class AzureBlobStorage implements Adapter,
     protected $blobProxy;
 
     /**
+     * @var bool
+     */
+    protected $multiContainerMode = false;
+
+    /**
+     * @var CreateContainerOptions
+     */
+    protected $createContainerOptions;
+
+    /**
      * @param AzureBlobStorage\BlobProxyFactoryInterface $blobProxyFactory
-     * @param string                                     $containerName
+     * @param string|null                                $containerName
      * @param bool                                       $create
      * @param bool                                       $detectContentType
+     *
+     * @throws \RuntimeException
      */
-    public function __construct(BlobProxyFactoryInterface $blobProxyFactory, $containerName, $create = false, $detectContentType = true)
+    public function __construct(BlobProxyFactoryInterface $blobProxyFactory, $containerName = null, $create = false, $detectContentType = true)
     {
         $this->blobProxyFactory = $blobProxyFactory;
         $this->containerName = $containerName;
         $this->detectContentType = $detectContentType;
-        if ($create) {
+        if (null === $containerName) {
+            $this->multiContainerMode = true;
+        } elseif ($create) {
             $this->createContainer($containerName);
         }
     }
 
     /**
+     * @return CreateContainerOptions
+     */
+    public function getCreateContainerOptions()
+    {
+        return $this->createContainerOptions;
+    }
+
+    /**
+     * @param CreateContainerOptions $options
+     */
+    public function setCreateContainerOptions(CreateContainerOptions $options)
```

Comment:
IMHO it would be better to have this as constructor parameter, it might even replace current `$create` argument: it could either take a `CreateContainerOptions` instance, meaning container(s) should be created if needed, or null, meaning it should fail if container(s) does not exist. But it would be a BC change (actually it would be better in another PR). What do you think?